### PR TITLE
Remove the removal of broken detached parts

### LINKED ISSUE #55174: Remove the removal of broken detached parts
This is a harmful feature - it has to be removed.
If there are broken parts, it requires immediate attention of an engineer.
They should never be deleted.

I remember this feature was added by one customer because they asked too many times.
But the truth is - the introduction of this feature is harmful even for that customer.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
There is no longer an option to automatically remove broken data parts. This closes #55174.

This writes off the previous work in #37975 #41190 #48730 #53164

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit d8b5713023843513f3c0cdbee6e3768eb7339f52 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55184/d8b5713023843513f3c0cdbee6e3768eb7339f52/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

**Comment by azat** [comment]
JFYI I was using this on one of clusters, because of on abnormal server restart, that was not that rare (due to faulty hardware) and without fsync it is highly lightly that you will have some broken parts after this. And usually in such parts all files are empty (but not always), and using automatic clean + `max_suspicious_broken_parts_bytes=1` may be not that bad.

Though I do agree that it is a dangerous feature.

File: tests/integration/test_broken_detached_part_clean_up/test.py [hunk 1]
```diff
@@ -1,360 +0,0 @@
...
-        node1.query(
-            f"INSERT INTO replicated_mt SELECT toDate('2019-10-01'), number, number * number FROM numbers ({i * 100000}, 100000)"
-        )
-
-    remove_broken_detached_part_impl("replicated_mt", node1, "broken")
-
-
-def test_store_cleanup(started_cluster):
-    node1.query("CREATE DATABASE db UUID '10000000-1000-4000-8000-000000000001'")
-    node1.query(
```
Comment (tavplubix) [inline tests/integration/test_broken_detached_part_clean_up/test.py:174]: @alexey-milovidov, this test was testing a different feature, please add it back

Comment (tavplubix) [reply tests/integration/test_broken_detached_part_clean_up/test.py:174]: https://github.com/ClickHouse/ClickHouse/pull/55605

**Pull request merged**